### PR TITLE
SendMail Command Script

### PR DESCRIPTION
**Pull request by sader1992**
* **Addressed Issue(s)**: none

* **Server Mode**: both

* **Description of Pull Request**: 

so this just an idea for now
the codes not completed yet
but i want to know what you think about it first

my plan is making a mail command can send zeny and items in it

this code work for zeny but not yet the items
i want to find the best way to make the command more user friendly for the items 
i did think about 
`SendMail(strcharinfo(0), "title", "body", 5, <zeny>,<itemid>,<amount>,<itemid>,<amount>);`till 5 items
but it wont look good

so i am thinking of making 3 or more commands like this
```
SendMail.item(<itemid>,<amount>);
SendMail.item2(<item2 values>);
SendMail.zeny(<zeny>);
SendMail.send(strcharinfo(0), "title", "body", 5);
```
i don't know if this would work 
those command work like we give the attachment for the mail then we send it

### CODE REVIEW

File: src/map/mail.cpp [hunk 1]
```diff
@@ -418,3 +419,39 @@ void mail_refresh_remaining_amount( struct map_session_data* sd ){
...
 }
+
+void npc_mail_send(struct npc_data *nd, const char *dest_name, const char *title, const char *body_msg, int body_len, int zeny) {
+	struct mail_message msg;
+
+	if (body_len > MAIL_BODY_LENGTH)
+		body_len = MAIL_BODY_LENGTH;
+
+	msg.zeny = zeny;
+	//msg.item[1].nameid = 502;
```
Comment (anacondaqq) [inline src/map/mail.cpp]: typo?

File: src/map/mail.cpp [hunk 2]
```diff
@@ -418,3 +419,39 @@ void mail_refresh_remaining_amount( struct map_session_data* sd ){
 		sc_start2( &sd->bl, &sd->bl, SC_DAILYSENDMAILCNT, 100, doy, 0, -1 );
 	}
 }
+
+void npc_mail_send(struct npc_data *nd, const char *dest_name, const char *title, const char *body_msg, int body_len, int zeny) {
```
Comment (anacondaqq) [inline src/map/mail.cpp]: 1. Why VOID? Why not to try to use ENUMs with named constants which will signal that something wrong?
2. suggestion: const char body_msg to just msg, and body_len to msg_len

File: src/map/mail.cpp [hunk 3]
```diff
@@ -418,3 +419,39 @@ void mail_refresh_remaining_amount( struct map_session_data* sd ){
...
+
+	msg.zeny = zeny;
+	//msg.item[1].nameid = 502;
+	//msg.item[1].amount = 1;
+	for (int i = 0; i < MAIL_MAX_ITEM; i++) {
+		msg.item[i].nameid = 0;
+		msg.item[i].amount = 0;
+	}
+	msg.id = 0; // id will be assigned by charserver
+	msg.send_id = nd->bl.id;
```
Comment (anacondaqq) [inline src/map/mail.cpp]: add nullpointer check for nd->bl.id before usage.

File: src/map/mail.cpp [hunk 4]
```diff
@@ -418,3 +419,39 @@ void mail_refresh_remaining_amount( struct map_session_data* sd ){
...
+	msg.dest_id = 0; // will attempt to resolve name
+	safestrncpy(msg.send_name, nd->name, NAME_LENGTH);
```
Comment (anacondaqq) [inline src/map/mail.cpp]: nd->name nullpointer check before usage prefered.

File: src/map/mail.cpp [hunk 5]
```diff
@@ -418,3 +419,39 @@ void mail_refresh_remaining_amount( struct map_session_data* sd ){
...
+	safestrncpy(msg.dest_name, (char*)dest_name, NAME_LENGTH);
+	safestrncpy(msg.title, (char*)title, MAIL_TITLE_LENGTH);
+	msg.type = MAIL_INBOX_NORMAL;
```
Comment (anacondaqq) [inline src/map/mail.cpp]: how about adding this to parameters list?
The whole function will be agiler if will be added the type options to the parameters...
So people can decide what type of mail they wish to send 🍅

File: src/map/mail.cpp [hunk 6]
```diff
@@ -418,3 +419,39 @@ void mail_refresh_remaining_amount( struct map_session_data* sd ){
...
+
+	if (msg.title[0] == '\0') {
+		return; // Message has no length and somehow client verification was skipped.
+	}
+
+	if (body_len)
+		safestrncpy(msg.body, (char*)body_msg, body_len + 1);
+	else
+		memset(msg.body, 0x00, MAIL_BODY_LENGTH);
```
Comment (anacondaqq) [inline src/map/mail.cpp]: is not better replace it with `\0`? (I can be wrong)

File: src/map/script.cpp [hunk 7]
```diff
@@ -24376,6 +24392,8 @@ struct script_function buildin_func[] = {
 	BUILDIN_DEF2(round, "ceil", "i"),
 	BUILDIN_DEF2(round, "floor", "i"),
 	BUILDIN_DEF(getequiptradability, "i?"),
+	
+	BUILDIN_DEF(sendmail, "?????"),
```
Comment (anacondaqq) [inline src/map/script.cpp]: why?
id, id, str, str, str, num, num
Why ??????  then? It can be a little bit confused.

Just suggestion:
iisssii

**Comment by Daegaladh** [comment]
I suggest to name it just mail, since can be confusing with the sendmail commands in other languages that send actual email.

**Comment by Lemongrass3110** [comment]
As discussed with @sader1992 in private chat.

**Pull request closed without merging**